Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py

```diff
@@ -266,6 +274,15 @@ def validateInputs(self):
             if self.getProperty("SampleFormula").value == '':
                 issues['SampleFormula'] = "A sample formula must be provided."
 
+        # The provided cache directory does not exist
+        cache_dir = self.getProperty('CacheDir').value  # absolute or relative path, as a string
+        if bool(cache_dir) and Path(cache_dir).exist() is False:
+            issues['CacheDir'] = f'Directory {cache_dir} does not exist'
+
+        # We cannot clear the cache if property "CacheDir" has not been set
+        if self.getProperty('CleanFile').value and not bool(self.getProperty('CacheDir').value):
+            issues['CleanFile'] = f'Property "CacheDir" must be set in order to clean the cache'
```

Comment:
Two questions here:

- Is "CleanFile" supposed to be "CleanCache"?
- Is this check necessary with the following policy in-place?
```
self.setPropertySettings('CleanCache', EnabledWhenProperty('CacheDir', PropertyCriterion.IsNotDefault))
```

Reply:
My reasoning was that `EnabledWhenProperty` only affects the algorithm's GUI. If you're using the algorithm in a script,`EnabledWhenProperty` will not be enforced, hence the extra validation.

Reply:
I didn't realize that this `EnableWhenProperty` is only useful for the GUI part :sweat_smile: